src/plugins/telemetry/server/plugin.ts

```diff
@@ -133,6 +134,10 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       logger: this.logger,
     });
 
+    // Only generate and report the snapshot telemetry in the UI node.
+    // This allows better cache optimizations and allowing background tasks to focus on alerts and similar.
+    this.shouldStartSnapshotTelemetryFetcher = initializerContext.node.roles.ui;
```

Comment:
What are the risks involved with _not_ collecting snapshot data from the background tasks node? Won't we be losing telemetry on task usage?

Do we have any indication of how much CPU we'd be freeing up?

Reply:
> What are the risks involved with _not_ collecting snapshot data from the background tasks node? Won't we be losing telemetry on task usage?

This is about gathering all the stats and shipping them. The collection is still happening and is being persisted so that it can be retrieved at this stage.

> Do we have any indication of how much CPU we'd be freeing up?

It varies: for small clusters it's unnoticeable. Larger ones might be impacted by longer collection times and processing larger datasets (with a lot of JSON parsing in the process), leading to high ELUs. We don't have exact numbers. But blaming Snapshot Telemetry's use of resources is a recurring theme.